ndsl/dsl/dace/orchestration.py

```diff
@@ -99,34 +88,39 @@ def _to_gpu(sdfg: dace.SDFG):
         else:
             arr.storage = dace.StorageType.GPU_Global
 
-    # All maps will be scedule on GPU
-    for mapentry, state in topmaps:
+    # All maps will be schedule on GPU
+    for mapentry, _state in topmaps:
         mapentry.schedule = dace.ScheduleType.GPU_Device
 
     # Deactivate OpenMP sections
     for sd in sdfg.all_sdfgs_recursive():
         sd.openmp_sections = False
 
 
-def _simplify(sdfg: dace.SDFG, validate=True, verbose=False):
+def _simplify(
+    sdfg: dace.SDFG,
+    *,
+    validate: bool = True,
+    validate_all: bool = False,
+    verbose: bool = False,
+):
     """Override of sdfg.simplify to skip failing transformation
     per https://github.com/spcl/dace/issues/1328
     """
     return SimplifyPass(
         validate=validate,
+        validate_all=validate_all,
```

Comment:
- `validate` validates the sdfg after all transformation have been applied
- `validate_all` validates after each transformation. Useful for debugging.

Reply:
`ConstantPropagation` should be passing now. Just a note.

I am afraid of that `validate_all` dying on the full dynamical core but it's fair game

Reply:
I logged an issue to re-evaluate skipping `ConstantPropagation` in https://github.com/GEOS-ESM/NDSL/issues/68.

The default for `validate_all` is `False`. It just makes people's lives easier when debugging because you can optionally pass this along to our `_simplify()`. Nothing will break from having that option. It's there (and also defaults to `False`) in DaCe's version of `simplify()` too.